Sources/AdcioCore/Feature/AdcioCore.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+import UIKit
+
+class AdcioCore {
+    
+    public static let shared = AdcioCore()
+    
+    private init() {}
+    
+    private var isInitialized: Bool = false
+    private var clientIdValue: String = ""
+    private var sessionIdValue: String = ""
+    private var deviceIdValue: String = ""
+
+    public func initializeApp(
+        clientId newClientId: String,
+        sessionId newSessionId: String? = nil,
+        deviceId newDeviceId : String? = nil
+    ) {
+        self.isInitialized = true
+        self.clientIdValue = newClientId
+        self.sessionIdValue = newSessionId ?? UUID().uuidString
+        self.deviceIdValue = newDeviceId ?? fetchDefaultDeviceId()
+    }
+    
+    public var clientId: String {
+        get throws {
+            try ensureInitialized()
+            return self.clientIdValue
+        }
+    }
+    
+    public var storeId: String {
+        get throws {
+            try ensureInitialized()
+            return self.clientIdValue
+        }
+    }
+
+    public func sessionId(_ newId: String = "") throws -> String {
```

Comment:
전 이거 제가 잘 몰라서 그런걸 수도 있는데
String? 이렇게 하면 안되는건가요?

default값이 null이 아니라 empty라서 empty인것도 체크할 것 같긴한데 empty 보단 null로 체크가 더 정책으로 맞지 않나? 싶어서요

```suggestion
    public func sessionId(_ newId: String?) throws -> String {
```

Reply:
아하! 저번에 Swift 스러운 코드? 에 대해서 공부할 때 isEmpty 함수 사용을 권장하더라고요!
근데 nullable은 isEmpty 처리할 때 좀 더 코드가 복잡해져서 저런식으로 구현하였습니다. 그리고 ?로 하면 파라미터가 optional 파라미터가 되지 않을 것 같아요.

Reply:
> 그리고 ?로 하면 파라미터가 optional 파라미터가 되지 않을 것 같아요.
nullable이 optional 파라미터가 안된다는게 의문 이네요? 확실한 내용이 맞나요? 한번 관련 된 내용을 참고할 만한 링크 있다면 첨부해주세요 @aahspringaa4 

> 그리고 nullable값 대신 empty 체크가 맞는지?
흠 이건 swift 언어를 같이 공부하신 @snack655 님이 한번 더블체크해주시면 좋을 것 같아요
저는 그러면 지금 코드도 동작 상 맞는 것 같으니 민재님께 책임을 남기고 전 approve 하겠습니당

Reply:
isEmpty가 왜 Swift스러운 코드인지? 이유가 좀 부족한 것 같아요.

사실 Optional이든 공백?("") 이든 같은 동작을 할 수 있지만 구현 방식이 다르다고 생각해요.
왜 isEmpty가 Swift 스러운 코드인지는 잘 모르겠지만
**공백("", isEmpty)으로 빈 값 검사를 하면 error을 야기할 수 있는 nil을 사용하지 않아서 확실히 더 안전한거 같아요**

Reply:
제가 한 말을 약간 정리하면 
new: String? 형태가 optional 파라미터가 아닌 거고! new: String? = nil 은 Optional 파라미터가 되긴합니다!
근데 nullable 인경우 isEmpty 사용에 어려움이 있어 non nullable로 정의 하였습니다! 

IsEmpty에 관한 참고자료 붙혀놓겠습니다!

https://iosmason.notion.site/Performance-functional-programming-and-collections-in-Swift-93c1101572b04e2b921dc5dfe083bdde